lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+        let offset = self.offset.consume();
+        let escape = Token::TextSegmentEscape(token::EscapeStyle::Literal,"'",offset);
+        self.discard_current();
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering an escaped interpolate quote.
+    fn on_escape_interpolate_quote<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_interpolate_quote");
+        let offset = self.offset.consume();
+        let escape = Token::TextSegmentEscape(token::EscapeStyle::Literal,"`",offset);
+        self.discard_current();
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering an escaped backslash.
+    fn on_escape_slash<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_slash");
+        let offset = self.offset.consume();
+        let escape = Token::TextSegmentEscape(token::EscapeStyle::Literal,"\\",offset);
+        self.discard_current();
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering an unrecognized escape sequence.
+    fn on_escape_invalid<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_invalid");
+        let offset  = self.offset.consume();
+        let current = self.consume_current();
+        let escape  = Token::TextSegmentEscape(token::EscapeStyle::Invalid,current,offset);
+        self.text_state.append_segment(escape);
+    }
+
+    // === Rule Definitions ===
+
     /// Define the rules for lexing Enso text literals.
-    fn add_text_rules(_lexer:&mut EnsoLexer) {
-        // TODO [AA] Write the lexing rules for text literals.
+    fn add_text_rules(lexer:&mut EnsoLexer) {
+        let format_quote         = c!('\'');
+        let format_block_quote   = l!("'''");
+        let raw_quote            = c!('"');
+        let raw_block_quote      = l!("\"\"\"");
+        let backslash            = c!('\\');
+        let spaces               = EnsoLexer::spaces();
+        let eof                  = Pattern::eof();
+        let interpolate_quote    = c!('`');
+        let lf                   = l!("\n");
+        let crlf                 = l!("\r\n");
+        let format_block_lf      = &format_block_quote >> spaces.opt() >> &lf;
+        let format_block_crlf    = &format_block_quote >> spaces.opt() >> &crlf;
+        let raw_block_lf         = &raw_block_quote >> spaces.opt() >> &lf;
+        let raw_block_crlf       = &raw_block_quote >> spaces.opt() >> &crlf;
+        let format_char          = Pattern::none_of("'`\\\n\r");
+        let format_raw_segment   = format_char.many1();
+        let format_block_segment = Pattern::none_of("\n\r\\`").many1();
+        let raw_block_segment    = Pattern::none_of("\n\r\\");
+        let raw_segment          = Pattern::none_of("\"\n\r\\").many1();
+        let unicode_escape_char  = Pattern::none_of("'\"`\\\n\r{}");
```

Comment:
why we are repeating `\r\n` so many times in this file? And why `raw_block_segment` is defined as newline followed by something? Basically the names around this place seem very strange to me